src/thermo/BinarySolutionTabulatedThermo.cpp

```diff
@@ -0,0 +1,230 @@
+/**
+ *  @file BinarySolutionTabulatedThermo.cpp Implementation file for an binary solution model
+ *      with tabulated standard state thermodynamic data (see \ref thermoprops and
+ *      class \link Cantera::BinarySolutionTabulatedThermo BinarySolutionTabulatedThermo\endlink).
+ */
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at http://www.cantera.org/license.txt for license and copyright information.
+
+#include "cantera/thermo/BinarySolutionTabulatedThermo.h"
+#include "cantera/thermo/PDSS.h"
+#include "cantera/thermo/ThermoFactory.h"
+#include "cantera/base/stringUtils.h"
+#include "cantera/base/ctml.h"
+#include "cantera/thermo/SpeciesThermoFactory.h"
+#include "cantera/thermo/MultiSpeciesThermo.h"
+
+namespace Cantera
+{
+
+BinarySolutionTabulatedThermo::BinarySolutionTabulatedThermo(int formGC) :
+    m_formGC(formGC),
+    m_Pref(OneAtm),
+    m_Pcurrent(OneAtm)
+{
+    if (formGC < 0 || formGC > 2) {
+        throw CanteraError(" BinarySolutionTabulatedThermo Constructor",
+                           " Illegal value of formGC");
+    }
+}
+
+BinarySolutionTabulatedThermo::BinarySolutionTabulatedThermo(const std::string& inputFile,
+        const std::string& id_, int formGC) :
+    m_formGC(formGC),
+    m_Pref(OneAtm),
+    m_Pcurrent(OneAtm)
+{
+    if (formGC < 0 || formGC > 2) {
+        throw CanteraError(" BinarySolutionTabulatedThermo Constructor",
+                           " Illegal value of formGC");
+    }
+    initThermoFile(inputFile, id_);
+}
+
+BinarySolutionTabulatedThermo::BinarySolutionTabulatedThermo(XML_Node& root, const std::string& id_,
+        int formGC) :
+    m_formGC(formGC),
+    m_Pref(OneAtm),
+    m_Pcurrent(OneAtm)
+{
+    if (formGC < 0 || formGC > 2) {
+        throw CanteraError(" BinarySolutionTabulatedThermo Constructor",
+                           " Illegal value of formGC");
+    }
+    importPhase(root, this);
+}
+
+void BinarySolutionTabulatedThermo::compositionChanged()
+{
+	IdealSolidSolnPhase::compositionChanged();
+    _updateThermo();
+}
+
+void BinarySolutionTabulatedThermo::_updateThermo()

```

Comment:
Since the only place this function is called is the one location in `compositionChanged`, it would be simpler to just include the contents of this function in `compositionChanged` directly.

Reply:
The possible exception here is that the parent class (IdealSolidSolution) also calls _updateThermo().  It is not clear to me whether these calls from the parent class also need to check whether the composition has been changed.